Shared/API/CloudFunction.swift

```diff
@@ -27,8 +26,12 @@ extension CloudFunction {
                      viewsToIgnore: [UIView]) async throws -> Any {
 
         // Trigger the loading event for all statusables
-        for statusable in statusables {
-            statusable.handleEvent(status: .loading)
+        await withTaskGroup(of: Void.self) { group in
+            for statusable in statusables {
+                group.addTask {
+                    await statusable.handleEvent(status: .loading)
+                }
+            }
```

Comment:
Couple things to note here:

- Handle event is now an async function.

- This task group code allows us to run all the handle events calls simultaneously. It will resume once all calls are complete.